codex-rs/core/src/shell.rs

```diff
@@ -0,0 +1,125 @@
+use shlex;
+use std::process::Command;
+use whoami;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Shell {
+    Zsh(String),
+    Unknown,
+}
+
+impl Shell {
+    pub fn run_with_profile(&self, command: Vec<String>) -> Option<Vec<String>> {
+        match self {
+            Shell::Zsh(shell_path) => {
+                let mut result = vec![shell_path.clone(), "-c".to_string()];
+                if let Ok(joined) = shlex::try_join(command.iter().map(|s| s.as_str())) {
+                    result.push(format!("source ~/.zshrc && ({joined})"));
+                } else {
+                    return None;
+                }
+                Some(result)
+            }
+            Shell::Unknown => None,
+        }
+    }
+}
+
+#[cfg(target_os = "macos")]
+pub fn current_shell() -> Option<Shell> {
+    let user = whoami::username();
+    let output = Command::new("dscl")
```

Comment:
Consider `tokio::Command` and use `async` on the off chance invoking `dscl` gets wedged?